### PR TITLE
Add specify reason on join project request

### PR DESCRIPTION
**Pull request by samshara**
depends on https://github.com/the-deep/server/pull/490/

### CODE REVIEW

File: src/views/DiscoverProjects/Actions/JoinProjectModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                onChange={handleFormChange}
+                value={joinProjectFaramValues}
+                error={joinProjectFaramErrors}
+                onValidationSuccess={handleValidationSuccess}
+                onValidationFailure={handleValidationFailure}
+                disabled={projectJoinRequestPending}
+            >
+                <ModalBody className={styles.modalBody}>
+                    <div>
+                        {_ts('discoverProjects.joinProject', 'requestQuestion', { projectName })}
```
Comment (AdityaKhatri) [inline src/views/DiscoverProjects/Actions/JoinProjectModal/index.tsx]:
```suggestion
                        {_ts('discoverProjects.joinProject', 'requestQuestion', { 
                            projectName: (
                                <span className={styles.projectName}>
                                    {projectName}
                                </span>
                            ),
                         })}
```
lets do this and make the project title bold

File: src/views/DiscoverProjects/Actions/JoinProjectModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                    </div>
+                    <TextArea
+                        faramElementName="reason"
```
Comment (AdityaKhatri) [inline src/views/DiscoverProjects/Actions/JoinProjectModal/index.tsx:129]: Put in autofocus and rows="5" for added height

File: src/views/DiscoverProjects/Actions/JoinProjectModal/styles.scss [hunk 3]
```diff
@@ -0,0 +1,10 @@
+.modal {
+    width: 50vw;
+    min-width: 720px;
```
Comment (AdityaKhatri) [inline src/views/DiscoverProjects/Actions/JoinProjectModal/styles.scss]: Lets have min width 560px and max width 720px

File: src/views/DiscoverProjects/Actions/index.js [hunk 4]
```diff
@@ -21,18 +24,26 @@ const propTypes = {
...
     onProjectJoin: PropTypes.func.isRequired,
     onProjectJoinCancel: PropTypes.func.isRequired,
+    projectJoinRequestPending: PropTypes.bool.isRequired,
 };
 
 const defaultProps = {
     // className: '',
 };
 
+const ModalButton = modalize(Button);
```
Comment (AdityaKhatri) [inline src/views/DiscoverProjects/Actions/index.js]: Lets use PrimaryButton instead of Button

File: src/redux/initial-state/dev-lang.json [hunk 5]
```diff
@@ -1141,7 +1141,10 @@
         "-59901057": "Ary",
         "-25520086": "This lead is in the Assessment Registry.",
         "-27768788": "Lead successfully edited.",
-        "-39063797": "Published On"
+        "-39063797": "Published On",
+        "-11231231": "Send Request",
+        "-10909809": "Request to join ${projectName}",
```
Comment (AdityaKhatri) [inline src/redux/initial-state/dev-lang.json]: We don't need $ here :P

File: src/views/DiscoverProjects/Actions/JoinProjectModal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+            projectName,
+            body: faramValues,
+        });
+    };
+
+    const handleValidationFailure = (faramErrors: setJoinProjectFaramErrors) => {
+        setJoinProjectFaramErrors(faramErrors);
+    };
+
+    const schema: Schema = {
```
Comment (tnagorra) [inline src/views/DiscoverProjects/Actions/JoinProjectModal/index.tsx]: We should move this outside the component

**Review by AdityaKhatri — APPROVED** [review]

File: src/components/general/Navbar/Notifica/Notifications/items/EntryCommentItem/index.js [hunk 7]
```diff
@@ -152,6 +152,7 @@ function EntryCommentItem(props) {
             createdByDetail: {
                 displayPicture,
             } = {},
+            // text,
```
Comment (tnagorra) [inline src/components/general/Navbar/Notifica/Notifications/items/EntryCommentItem/index.js]: Why add these comments?

File: src/components/general/Navbar/Notifica/Notifications/items/EntryCommentItem/index.js [hunk 8]
```diff
@@ -172,6 +173,8 @@ function EntryCommentItem(props) {
             }
             message={notificationText}
             timestamp={timestamp}
+            // description={text}
+            // descriptionLabel={_ts('entryComments.notifications', 'expandButtonLabel')}
```
Comment (tnagorra) [inline src/components/general/Navbar/Notifica/Notifications/items/EntryCommentItem/index.js]: Why add these comments? Are they to be used in future?

Comment (AdityaKhatri) [reply src/components/general/Navbar/Notifica/Notifications/items/EntryCommentItem/index.js]: yes, I'll provide this work to you and you can do it quickly

File: src/components/general/Navbar/Notifica/Notifications/items/EntryCommentItem/index.js (hunk 7 above)
Comment (AdityaKhatri) [reply src/components/general/Navbar/Notifica/Notifications/items/EntryCommentItem/index.js]: TO do later on

**Review by tnagorra — APPROVED** [review]

**Review by tnagorra — APPROVED** [review]

**Review by tnagorra — APPROVED** [review]

**Pull request merged**